src/components/artefact/artefact-svg.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+    <div v-show="loaded">
+        <svg v-if="loaded"
+            :viewBox="`${boundingBox.x} ${boundingBox.y} ${boundingBox.width} ${boundingBox.height}`"
+            :width="elementWidth"
+            :height="elementHeight" 
+            xmlns="http://www.w3.org/2000/svg"
+            xmlns:xlink="http://www.w3.org/1999/xlink">
+            <defs>
+                <path
+                    :id="`path-${artefact.id}`"
+                    :d="artefact.mask.polygon.svg"/>
+                <clipPath :id="`clip-path-${artefact.id}`">
+                    <use stroke="none" fill="none" fill-rule="evenodd" :xlink:href="`#path-${artefact.id}`"></use>
+                </clipPath>
+            </defs>
+
+            <g :clip-path="`url(#clip-path-${artefact.id}`">
+                <slot v-bind:getImageUrl="getImageUrl"></slot>
+            </g>
+        </svg>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue, Mixins } from 'vue-property-decorator';
+import { Artefact } from '@/models/artefact';
+import { ImageStack, IIIFImage } from '@/models/image';
+import { Polygon } from '@/utils/Polygons';
+import { BoundingBox } from '@/utils/helpers';
+import { ImageSetting, SingleImageSetting } from '@/components/image-settings/types';
+import ArtefactDataMixin from './artefact-data-mixin';
+
+@Component({
+    name: 'artefact-svg',
+})
+export default class ArtefactSvg extends  Mixins(ArtefactDataMixin) {
+    @Prop({default: 1.3}) private aspectRatio!: number;
+
+    private elementWidth = 0;
+    private serverScale = 5;
+    private loaded = false;
+
+    get scale(): number {
+        if (this.elementWidth && this.masterImageManifest) {
+            return this.elementWidth / this.boundingBox.width;
+        }
+
+        return 0.05;
+    }
+
+    get elementHeight(): number {
+        if (this.elementWidth) {
+            return this.elementWidth / this.aspectRatio;
+        }
+
+        return 100;
+    }
+
+    protected async mounted() {
```

Comment:
I needed to figure out how to handle an async mounted in the mixin. Sync mounted calls are handled properly - mixins are called before the component. With asyncs, I needed to add an `await` here. Since mounted is already called automatically, adding another function that we need to remember to call seemed like a bit of a downgrade.

Reply:
Ok, fine with me.